lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -1039,3 +1041,8 @@ invalid_do_error(Prefix) ->
   "    if(some_condition?, do: :this, else: :that)\n\n"
   "where \"some_condition?\" is the first argument and the second argument is a keyword list.\n\n"
   "Syntax error before: ".
+
+do_with_fn_error(Prefix) ->
+  Prefix ++ ". Anonymous function literals do not accept do...end blocks. Their syntax is:\n\n"
```

Comment:
I think the accept "do...end" does not read well. Given we already said we errored on `do`, we should just say "Anonymous functions are written as `fn pattern ->expr end`. Syntax error before: "?